src/components/formControl/index.tsx

```diff
@@ -79,8 +94,11 @@ const FormControl: FC<PropsWithChildren<Props>> = ({
       id={id}
     >
       {label && !tooltip && !labelButtonText ? formLabel : null}
-      {label && tooltip ? formLabelWithTooltip : null}
-      {label && labelButtonText ? formLabelWithButton : null}
+      {label && tooltip && !labelButtonText ? formLabelWithTooltip : null}
```

Comment:
suggestion, this is very verbose and could result in impossible states where you'd show multiple things. Since label must always be defined, can we do something like:

```ts
let labelComponent = null;

if (label) {
  if (tooltip && labelButtonText) {
    labelComponent = formLabelWithTooltipAndButton;
  } else if (tooltip) {
    labelComponent = formLabelWithTooltip;
  } else if (labelButtonText) {
    labelComponent = formLabelWithButton;
  } else {
    labelComponent = formLabel;
  }
}

// Use the labelComponent here
{labelComponent}
```